src/EventsCapable/EventsCapableInitializer.php

```diff
@@ -47,11 +48,36 @@ public function initialize($instance, ServiceLocatorInterface $serviceLocator)
         
         $listeners = $options->getListeners($instance);
         
-        foreach ($listeners as $listenerName) {
-            $listener = $serviceLocator->get($listenerName);
+        foreach ($listeners as $listener) {
+            if (!$listener instanceof ListenerAggregateInterface) {
```

Comment:
You should probably ensure this is a string as any object that doesn't implement ListenerAggregateInterface won't cause an error later on when doing your service lookup